packages/wallet/src/mock-rollup-state-machine.ts

```diff
@@ -0,0 +1,209 @@
+/* External Imports */
+
+/* Internal Imports */
+import {
+  Address,
+  Balances,
+  Swap,
+  isSwapTransaction,
+  Transfer,
+  isTransferTransaction,
+  Transaction,
+  MockedSignature,
+  SignedTransaction,
+  TransactionReceipt,
+  UNISWAP_ADDRESS,
+  UNI_TOKEN_TYPE,
+  PIGI_TOKEN_TYPE,
+  TokenType,
+  State,
+} from '.'
+
+const DEFAULT_STORAGE = {
+  balances: {
+    uni: 0,
+    pigi: 0,
+  },
+}
+
+/*
+ * Errors
+ */
+export class SlippageError extends Error {
+  constructor() {
+    super('Too much slippage in swap tx!')
+  }
+}
+
+export class InsufficientBalanceError extends Error {
+  constructor() {
+    super('Insufficient balance for transfer or swap!')
+  }
+}
+
+export class NegativeAmountError extends Error {
+  constructor() {
+    super('Amounts transferred or swapped cannot be negative!')
+  }
+}
+
+export class InvalidTransactionTypeError extends Error {
+  constructor() {
+    super('Invalid transaction type!')
+  }
+}
+
+export class MockRollupStateMachine {
+  public state: State
+
+  constructor(genesisState: State) {
+    this.state = genesisState
+  }
+
+  public getBalances(account: Address): Balances {
+    if (!(account in this.state)) {
```

Comment:
If you flip this to 
```
if (!(account in this.state)) {
  this.state[account] = {
    uni: 0,
    pigi: 0
  }
}

return this.state[account]
```
you can just call `this.getBalances(...)` for all balance access and not worry about whether or not the account exists.